### DISCUSSION TITLE
ALT tag should be easier to access

### DISCUSSION THREAD

**Discussion by vaughn-taylor** [craftcms/cms#14893] [Ideas] [answered]
With more people caring about ADA (or requiring it), and the SEO benefit of ALT text, it's sure not easy to access when adding an asset. For example, when I add an image to my Matrix gallery (in card view), I need to click the image twice before I can edit the ALT text:

<img width="1226" alt="Screenshot 2024-04-29 at 11 15 39 AM" src="https://github.com/craftcms/cms/assets/2078376/9d680ee7-2bc0-4b23-9d4d-953237931c0e">

Ideally, the ALT field would be present and editable beside the image. Something like this would be ideal:

<img width="444" alt="Screenshot 2024-04-29 at 11 19 23 AM" src="https://github.com/craftcms/cms/assets/2078376/e26008bf-8c3b-42da-8c2a-a03cef69931c">

**Comment by thupsi** [comment]
FWIW you can configure alt text to display right in the asset card by enabling the option `show in element cards` in the alt text field. It will not be editable inline, but it will be visible. When the "Preparse" plugin is available for Craft 5 you could even use this to display "no alt text for this image" or something along those lines, when alt text is empty.

**Comment by lukew-cogapp** [comment]
+1 from us!

**Comment by brandonkelly** [comment, marked as answer]
Alt text is stored globally for the asset, but adding the ability to edit it inline within the card would give the impression that it’s customizable on a per-relationship basis. So I don’t think we should do that.

**Reply by lukew-cogapp** [reply]
What about a warning that the selected asset has a missing required field? Even better if it told you the field. This would then be generic, but very useful for alt text. 

The issue is that required fields on assets aren't checked on upload, only when the asset is next saved.

It's forever going to be a struggle to get users to add alt text, we've had to resort to adding a separate field to the entry in the past, but of course as soon as it's used elsewhere, this info would have to be duplicated.

**Reply by vaughn-taylor** [reply]
It's critical and required for ADA. I'd love to see it as an editable field directly in list view on the Assets page, at very least.

**Reply by brandonkelly** [reply]
@lukew-cogapp You can require that assets’ alt text is filled in before they can be related to entries, by A) ensuring that you’ve marked the Alternative Text field as required within the volume’s field layout; and B) enabling your Assets field’s “Validate related assets” setting.

@vaughn-taylor Good call. I’ve just made it possible to edit alt text from the Assets index page for Craft 5.2 (aa4d6db7a8c4485853e24fefae513f18fc19eaff).

**Reply by brandonkelly** [reply]
Craft 5.2.0-beta.1 is out with that change.

**Reply by ryanleichty** [reply]
I know this issue has already been resolved, but [a somewhat related feature that Sanity has](https://www.sanity.io/docs/validation#812ebcd2ff64) is the ability to issue warnings as an alternative to making a field required. This is great for alt text where you might not want to prevent someone from saving an asset, but you still want to surface a message indicating that something needs to be corrected.

**Reply by brandonkelly** [reply]
@ryanleichty Mind posting that as a new discussion?
